google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/stub/metrics/BuiltinMetricsTracer.java

```diff
@@ -239,26 +274,42 @@ private void recordOperationCompletion(@Nullable Throwable status) {
       return;
     }
     operationTimer.stop();
-    long operationLatency = operationTimer.elapsed(TimeUnit.MILLISECONDS);
+
+    boolean isStreaming = operationType == OperationType.ServerStreaming;
+    String statusStr = Util.extractStatus(status);
+
+    Attributes attributes =
+        baseAttributes
+            .toBuilder()
+            .put(TABLE_ID_KEY, tableId)
+            .put(CLUSTER_ID_KEY, cluster)
+            .put(ZONE_ID_KEY, zone)
+            .put(METHOD_KEY, spanName.toString())
+            .put(CLIENT_NAME_KEY, NAME)
+            .build();
+
     long operationLatencyNano = operationTimer.elapsed(TimeUnit.NANOSECONDS);
 
     // Only record when retry count is greater than 0 so the retry
     // graph will be less confusing
     if (attemptCount > 1) {
-      recorder.putRetryCount(attemptCount - 1);
+      retryCounter.add(attemptCount - 1, attributes.toBuilder().put(STATUS_KEY, statusStr).build());
     }
 
     // serverLatencyTimer should already be stopped in recordAttemptCompletion
-    recorder.putOperationLatencies(operationLatency);
-    recorder.putApplicationLatencies(
-        Duration.ofNanos(operationLatencyNano - totalServerLatencyNano.get()).toMillis());
+    operationLatenciesHistogram.record(
+        Duration.ofNanos(operationLatencyNano).toMillis(),
```

Comment:
I think you forgot to update this  to a double? (everything else uses your toMs multiplier)

Also, please extract the multiplication into a named function and add documentation as to why it is needed